daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/ValueGenerators.scala

```diff
@@ -323,7 +326,8 @@ object ValueGenerators {
         stakeholders,
         signatories,
         controllers,
-        children
+        children,
+        None // Some(exerciseResultValue) TODO - See above
```

Comment:
So, as you've surmised, `Some(exerciseResultValue)` is the right thing to have here. But then you have the problem that this gets discarded for encoding transactions < version `minExerciseResult`, right?

For the roundtrip tests you need to use a looser comparison. You can just drop the `exerciseResult` from each side:

```scala
def withoutExerciseResult[Nid, Cid, Val](gn: GenNode[Nid, Cid, Val])
    : GenNode[Nid, Cid, Val] =
  gn match {
    case ne: NodeExercises[Nid, Cid, Val] => ne copy (exerciseResult = None)
    case _ => gn
  }

def withoutExerciseResults[Nid, Cid, Val](t: GenTransaction[Nid, Cid, Val])
    : GenTransaction[Nid, Cid, Val] =
  t copy (nodes = t.nodes transform ((_, gn) => withoutExerciseResult(gn)))
```

Then you can say something like

```scala
  roundtripped shouldBe (if (version precedes minExerciseResult)
                           withoutExerciseResults(original)
                         else original)
```

Reply:
Done